src/braket/experimental/algorithms/quantum_partition_function/quantum_partition_function.py

```diff
@@ -0,0 +1,202 @@
+        Ga = potts_model["graph-model"]
+        # q = potts_model["q-state"]
+        # nodes of graph
+        N = len(Ga.nodes)
+        # edges of graph
+        E = len(Ga.edges)
+        # the number of connected components in the graph
+        C = len([Ga.subgraph(c).copy() for c in nx.connected_components(Ga)])
+        # [n,k] code
+        n = N
+        k = E - C
+        out["nk-code"] = (n, k)
+        out["connected-component"] = C
+    elif step == "iccc-check":
+        print("Irreducible Cyclic Cocycle Code Check")
+        if "nk-code" not in potts_model.keys():
+            raise KeyError("no nk-code found in potts_model, please run pre-process step!")
+        Ga = potts_model["graph-model"]
+        # TODO: Add shor algorithm to check ICCC
+        N = len(Ga.nodes)
+        E = len(Ga.edges)
+        n, k = potts_model["nk-code"]
+        C = potts_model["connected-component"]
+        print(f"The cycle matroid matrix of Graph Gamma is {n-C} x {n}")
+        check_result = False
+        if N == 3 and E == 3:
+            print("the ICCC is [1,-1], which passes ICCC check!")
+            check_result = True
+        out["iccc-check"] = check_result
+    elif step == "qft":
+        print("State Preparation and Quantum Fourier Transform")
+        if "iccc-check" not in potts_model.keys():
+            raise KeyError("no iccc-check found in potts_model, please run ICCC-check step!")
+        check_result = potts_model["iccc-check"]
+        assert check_result is True, "ICCC check didn't pass, no efficient quantum alogrithm!"
+
+        if "qft-func" not in potts_model.keys():
+            raise KeyError("no qft-func found in potts_model, please generate circuit!")
+
+        qft_circuit = potts_model["qft-func"]["circuit"]
+        # Add desired results_types
+        qft_circuit.probability()
+
+        qft_param = potts_model["qft-func"]["param"]
+        qft_device = qft_param["device"]
+        qft_shots = qft_param["shots"]
+
+        task = qft_device.run(qft_circuit, shots=qft_shots)
+
+        circ_result = {
+            "task": task,
+        }
+
+        out.update(circ_result)
+    elif step == "post":
+        print("Classical Post-Processing")
+        # TODO Classical pots process logic
+    else:
+        raise ValueError(
+            f"Wrong step value {step}, can only be one \
+                        of pre,ICCC-check,qft and post"
+        )
+
+    return out
+
+
+def get_quantum_partition_function_results(potts_model: Dict[str, Any]) -> None:
+    """Function to postprocess dictionary returned by run_quantum_partition_function and pretty
+        print results.
+
+    Args:
+        potts_model (dict): Dictionary to save the results of Potts model:
+            'graph-model': networkx graph to save the definition of potts model
+            'q-state': q-state of potts model
+            'qft-func': dictionary to save information for quantum fourier transform function
```

Comment:
```suggestion
            'qft-func': dictionary to save information for quantum Fourier transform function
```
I'll stop bothering with capitalization requests for the rest of the function, but there are still a few.